File: app/src/main/java/com/example/appestable/data/Persona.kt
package com.example.appestable.data

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "personas",
    foreignKeys = [
        ForeignKey(
            entity = Familia::class,
            parentColumns = ["id"],
            childColumns = ["familiaId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Viaje::class,
            parentColumns = ["id"],
            childColumns = ["viajeId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("familiaId"), Index("viajeId")]
)
data class Persona(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val nombre: String,
    val celular: String,
    val email: String,
    val familiaId: Int,
    val esJefe: Boolean,
    val backendId: Int? = null,
    val viajeId: Int = 1,
    val usuarioId: Int? = null,
    val rol: RolViaje = RolViaje.MIEMBRO
)

File: flutter_app/lib/models/persona.dart
import 'rol_viaje.dart';

class Persona {
  final int? id;
  final String nombre;
  final String? email;
  final String? celular;
  final bool esJefe;
  final int? familiaId;
  final String familiaNombre;
  final int? viajeId;
  final RolViaje rol;
  final String? auth0Id;

  Persona({
    this.id,
    required this.nombre,
    this.email,
    this.celular,
    required this.esJefe,
    this.familiaId,
    required this.familiaNombre,
    this.viajeId,
    this.rol = RolViaje.miembro,
    this.auth0Id,
  });

  factory Persona.fromJson(Map<String, dynamic> json) {
    int? _toInt(dynamic v) => v == null ? null : (v is num ? v.toInt() : int.tryParse(v.toString()));
    return Persona(
      id: _toInt(json['id']),
      nombre: json['nombre'] as String? ?? 'Sin nombre',
      email: json['email'] as String?,
      celular: json['celular'] as String?,
      esJefe: json['es_jefe'] as bool? ?? false,
      familiaId: _toInt(json['familia_id']),
      familiaNombre: json['familia_nombre'] as String? ?? 'Sin Familia',
      viajeId: _toInt(json['viaje_id']),
      rol: RolViaje.fromApi(json['rol'] as String?),
      auth0Id: json['auth0_id'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'email': email,
      'celular': celular,
      'es_jefe': esJefe,
      'familia_nombre': familiaNombre,
      if (auth0Id != null) 'auth0_id': auth0Id,
    };
  }
}